liferay-tango/liferay-portal
# LPS-101448 Collections Variations to be reordered for prioritization purposes while accessing the details

[Jira ticket](https://issues.liferay.com/browse/LPS-101448)

### Motivation
Variations have to be reorderable for prioritization, in case a user belongs to multiple segments.

### Proposed Solution
We are saving in db a priority value, that equals the index in the variations list.
The reordenation can be done via DnD, or vía click event.
The priority is saved every time the user alters the order of the list.
All the JSP has been migrated to React.

### Steps to Verify:
- Enable FF: `echo "enabled=B\"true\"" > ../bundles/osgi/configs/com.liferay.asset.list.web.internal.configuration.FFCollectionsVariationsPrioritizationConfiguration.config`
- Navigate to Site Builder > Collections
- Create new collection, manual or dynamic
- An initial state with only the default variation should be visible
- Finish the variation configuration
- Navigate to _Add new segment_ view
- Create one or more segments
- Navigate back to Collections
- Add the created segments to your Collection
- Change the order in the list and check everything get saved
![image (7)](https://user-images.githubusercontent.com/19485114/142824675-508eab15-278a-4672-8d3f-20b621f0362c.png)
![image (6)](https://user-images.githubusercontent.com/19485114/142824679-f2878621-3d90-472c-b93a-a5a729f6ec2e.png)
